### PR TITLE
[api][SIMS #1453]Created global Config and Logger modules.

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
- Moving LoggerService and ConfigService to the shared library.
- Created a module ConfigModule and declared it as `@Global` (global modules do not need to be imported in every single module).
- Logger module declared as `@Global`.
- ConfigService broke into smaller properties instead of a single method.

### CODE REVIEW

File: sources/packages/backend/apps/api/src/esdc-integration/sin-validation/sin-validation-processing.service.ts [hunk 1]
```diff
@@ -35,7 +30,7 @@ export class SINValidationProcessingService {
     private readonly sequenceService: SequenceControlService,
     private readonly sinValidationIntegrationService: SINValidationIntegrationService,
   ) {
-    this.esdcConfig = config.getConfig().ESDCIntegration;
+    this.esdcConfig = config.esdcIntegration;
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/esdc-integration/sin-validation/sin-validation-processing.service.ts:33]: so we no longer have this `getConfig` in config service?

Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/esdc-integration/sin-validation/sin-validation-processing.service.ts:33]: I don't think that we need it. The problem with the getConfig is that it parses all the env variables, which is not needed for other applications like `Workers` or the queue consumers that will be introduced. That means that not all env variables will be present on all PODs, so, depending on how they are read, the missing variables can cause an issue just for not being present.

File: sources/packages/backend/libs/utilities/src/config/config.service.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
+import { Injectable } from "@nestjs/common";
+import {
```
Comment (ann-aot) [inline sources/packages/backend/libs/utilities/src/config/config.service.ts:2]: 👍

File: sources/packages/backend/libs/utilities/src/logger/logger.service.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { Injectable, ConsoleLogger, Scope } from "@nestjs/common";
+
+/**
+ * Common log across entire solution.
+ */
+@Injectable({ scope: Scope.TRANSIENT })
```
Comment (ann-aot) [inline sources/packages/backend/libs/utilities/src/logger/logger.service.ts:6]: for my info what is `scope: Scope.TRANSIENT`

File: sources/packages/backend/libs/utilities/src/config/config.module.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { Global, Module } from "@nestjs/common";
+import { ConfigService } from "./config.service";
+
+@Global()
```
Comment (ann-aot) [inline sources/packages/backend/libs/utilities/src/config/config.module.ts:4]: for my info what does this decorator do?

File: sources/packages/backend/apps/api/src/main.ts [hunk 5]
```diff
@@ -59,25 +42,52 @@ async function bootstrap() {
...
+        req.headers.origin ?? "unknown"
+      } | ${req.headers["user-agent"] ?? "unknown"}`,
+    );
+    next();
+  });
 
   // pipes
   setGlobalPipes(app);
 
+  // Configure Swagger
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/main.ts:58]: fullstop

**Review by ann-aot — APPROVED** [review]
👍 Good work

File: sources/packages/backend/libs/utilities/src/logger/logger.service.ts (hunk 3 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/libs/utilities/src/logger/logger.service.ts:6]: From Nestjs docs: "Transient providers are not shared across consumers. Each consumer that injects a transient provider will receive a new, dedicated instance."
Basically, a new instance is created every time that it is injected.

File: sources/packages/backend/apps/api/src/main.ts (hunk 5 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/main.ts:58]: Not part of the PR.

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @andrewsignori-aot

**Pull request merged**